aregion.cpp

```diff
@@ -1293,25 +1386,32 @@ void ARegion::WriteExits(Areport *f, ARegionList *pRegs, int *exits_seen)
 "keep you safe as long as you should choose to stay. However, rumor " \
 "has it that once you have left the Nexus, you can never return."
 
-void ARegion::WriteReport(Areport *f, Faction *fac, int month,
+void ARegion::WriteReport(JsonReport &of, Faction *fac, int month,
 		ARegionList *pRegions)
 {
+	Areport *f = of.PlainFile();
 	Farsight *farsight = GetFarsight(&farsees, fac);
 	Farsight *passer = GetFarsight(&passers, fac);
 	int present = Present(fac) || fac->IsNPC();
 
 	if (farsight || passer || present) {
+		of.StartDict(NULL);
 		AString temp = Print(pRegions);
+		CPrint(&of, pRegions);
+
 		if (Population() &&
 			(present || farsight ||
 			 (Globals->TRANSIT_REPORT & GameDefs::REPORT_SHOW_PEASANTS))) {
 			temp += AString(", ") + Population() + " peasants";
+			of.PutPairInt("population", Population());
 			if (Globals->RACES_EXIST) {
 				temp += AString(" (") + ItemDefs[race].names + ")";
+				of.PutPairStr("race", ItemDefs[race].names);
 			}
 			if (present || farsight ||
 					Globals->TRANSIT_REPORT & GameDefs::REPORT_SHOW_REGION_MONEY) {
 				temp += AString(", $") + wealth;
+				of.PutPairInt("tax", wealth);
```

Comment:
we should probably put tax 0 in else condition too

Reply:
Fixed